test/manager_test.exs

```diff
@@ -84,124 +91,50 @@ defmodule ManagerTest do
   end
 
   test "Spider closed callback is called when spider is stopped" do
-    Process.register(self(), :spider_closed_callback_test)
     :ok = Crawly.Engine.start_spider(Manager.TestSpider)
     :ok = Crawly.Engine.stop_spider(Manager.TestSpider, :manual_stop)
-
-    assert_receive :manual_stop
+    assert :persistent_term.get(:spider_stop_reason) == :manual_stop
   end
 
   test "It's possible to start a spider with start_requests" do
     pid = self()
-    :ok = Crawly.Engine.start_spider(Manager.StartRequestsTestSpider)
-
-    :meck.expect(HTTPoison, :get, fn url, _, _ ->
-      send(pid, {:performing_request, url})
-
-      {:ok,
-       %HTTPoison.Response{
-         status_code: 200,
-         body: "Some page",
-         headers: [],
-         request: %{}
-       }}
-    end)
-
-    assert_receive {:performing_request, "https://www.example.com/blog.html"}
-  end
-
-  test "It's possible to initialize a spider with parameters" do
-    Process.register(self(), :manager_test_initial_args_test)
 
-    urls = [
-      "https://example.com/1",
-      "https://example.com/2",
-      "https://example.com/3"
-    ]
+    :meck.expect(Crawly.RequestsStorage, :store, fn _spider, request ->
+      send(pid, {:performing_request, request})
+    end)
 
-    :ok = Crawly.Engine.start_spider(Manager.InitialArgsTestSpider, urls: urls)
+    :ok =
+      Crawly.Engine.start_spider(Manager.StartRequestsTestSpider,
+        closespider_itemcount: :disabled,
+        closespider_timeout: :disabled
+      )
 
-    assert_receive recv_opts
-    assert is_binary(recv_opts[:crawl_id])
-    assert Enum.sort(recv_opts[:urls]) == Enum.sort(urls)
+    Process.sleep(100)
+    assert_receive {:performing_request, req}
+    assert req.url == "https://www.example.com/blog.html"
   end
 end
 
 defmodule Manager.TestSpider do
```

Comment:
I think instead of having a separate spider in the .exs test script, we should just mock the callbacks of the `TestSpider` module that is in the test_helpers.ex file, similar to how i did in the parse pipelines PR. it should make the tests less flaky.

Reply:
I have created a new ticket for that